src/assets/stylesheets/main/_api.scss

```diff
@@ -3,93 +3,96 @@ $api-color-keyword: var(--md-code-hl-keyword-color);
-        }
+  :is(dl.objdesc, dl.api-field) > dt {
+    font-family: var(--md-code-font-family);
+    background: $api-color-header-bg;
 
-        > a.reference:hover {
-          color: var(--md-accent-fg-color);
-        }
-      }
+    // Eliminate the normal padding for inline code blocks
+    code {
+      padding: 0;
+      border-radius: 0;
+    }
 
-      // Punctuation
-      .p {
-        color: $api-color-punctuation;
-      }
+    .sig-name:not(.sig-name-nonprimary) {
+      padding: 0;
+      color: $api-color-name;
+      font-weight: 700;
+    }
 
-      // Eliminate the normal padding for inline code blocks
-      code {
-        padding: 0;
-        border-radius: 0;
-      }
+    .sig-param {
+      font-style: normal;
+    }
 
-      .sig-name {
-        padding: 0;
-        color: $api-color-name;
-        font-weight: 700;
-      }
+    .sig-param .n:not(.desctype) {
+      color: $api-color-param-name;
+    }
 
-      a.reference > span {
-        color: var(--md-typeset-a-color);
+    .sig-param a.reference .n:not(.desctype):hover {
+      color: var(--md-accent-fg-color);
+    }
 
-        &:hover {
-          color: var(--md-accent-fg-color);
+    // Ensure each parameter starts on a new line and is indented.
+    &.sig-wrap {
+      .sig-param-decl {
+        &::before {
+          white-space: pre;
+          // 5 spaces below result in 4 spaces of indent.
+          // For some reason one space is lost.
+          content: "\A     ";
         }
       }
 
-      .sig-param {
-        font-style: normal;
+      .sig-paren:last-of-type {
+        // Ensure the final paren starts on a new line
+        &::before {
+          white-space: pre;
+          content: "\A";
+        }
       }
+    }
+  }
 
-      .sig-param .n:not(.desctype) {
-        color: $api-color-param-name;
-      }
+  // Rules for coloring cross-references in signatures.
+  :is(dl.objdesc, dl.api-field) > dt,
+  .sig-inline {
+    a.reference > .sig-name,
+    a.reference.sig-name,
+    a.reference:not(.desctype) > .n {
+      color: var(--md-typeset-a-color);
 
-      .sig-param a.reference .n:not(.desctype):hover {
+      &:hover {
         color: var(--md-accent-fg-color);
       }
+    }
 
-      // Ensure each parameter starts on a new line and is indented.
-      &.sig-wrap {
-        .sig-param-decl {
-          &::before {
-            white-space: pre;
-            // 5 spaces below result in 4 spaces of indent.
-            // For some reason one space is lost.
-            content: "\A     ";
-          }
+    .desctype {
+      &,
+      > a.reference {
+        color: $api-color-type;
+        // Prevent pygments highlight rules from taking precedence.
+        * {
+          color: inherit;
         }
+      }
 
-        .sig-paren:last-of-type {
-          // Ensure the final paren starts on a new line
-          &::before {
-            white-space: pre;
-            content: "\A";
-          }
-        }
+      &:hover,
```

Comment:
I think this selector is what is causing a non-hyperlinked return type look like hyperlink when hovered over (or tapped on in mobile browsers) as reported by [my other comment](https://github.com/jbms/sphinx-immaterial/pull/64/files#r855674950))